tabular/src/autogluon/tabular/models/tabpfn/tabpfn_model.py

```diff
@@ -0,0 +1,142 @@
+class TabPFNModel(AbstractModel):
+    """
+    AutoGluon model wrapper to the TabPFN model: https://github.com/automl/TabPFN
+
+    Paper: "TabPFN: A Transformer That Solves Small Tabular Classification Problems in a Second"
+    Authors: Noah Hollmann, Samuel Müller, Katharina Eggensperger, and Frank Hutter
+
+    TabPFN is a viable model option when inference speed is not a concern,
+    and the number of rows of training data is less than 10,000.
+
+    Additionally, TabPFN is only available for classification tasks with up to 10 classes and 100 features.
+
+    To use this model, `tabpfn` must be installed.
+    To install TabPFN, you can run `pip install autogluon.tabular[tabpfn]` or `pip install tabpfn`.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._feature_generator = None
+
+    def _fit(self, X: pd.DataFrame, y: pd.Series, **kwargs):
+        from tabpfn import TabPFNClassifier
+        ag_params = self._get_ag_params()
+        sample_rows = ag_params.get('sample_rows')
+        max_features = ag_params.get('max_features')
+        max_classes = ag_params.get('max_classes')
+        if max_classes is not None and self.num_classes > max_classes:
+            # TODO: Move to earlier stage when problem_type is checked
+            raise AssertionError(f'Max allowed classes for the model is {max_classes}, '
+                                 f'but found {self.num_classes} classes.')
+
+        # TODO: Make sample_rows generic
+        if sample_rows is not None and len(X) > sample_rows:
+            X, y = self._subsample_train(X=X, y=y, num_rows=sample_rows)
+        X = self.preprocess(X)
+        num_features = X.shape[1]
+        # TODO: Make max_features generic
+        if max_features is not None and num_features > max_features:
+            raise AssertionError(f'Max allowed features for the model is {max_features}, '
+                                 f'but found {num_features} features.')
+        hyp = self._get_model_params()
+        N_ensemble_configurations = hyp.get('N_ensemble_configurations')
+        self.model = TabPFNClassifier(
+            device='cpu',  # TODO: Add GPU option
+            N_ensemble_configurations=N_ensemble_configurations
+        ).fit(X, y, overwrite_warning=True)
+
+    # TODO: Make this generic by creating a generic `preprocess_train` and putting this logic prior to `_preprocess`.
+    def _subsample_train(self, X: pd.DataFrame, y: pd.Series, num_rows: int, random_state=0) -> (pd.DataFrame, pd.Series):
+        num_rows_to_drop = len(X) - num_rows
+        X, _, y, _ = generate_train_test_split(
+            X=X,
+            y=y,
+            problem_type=self.problem_type,
+            test_size=num_rows_to_drop,
+            random_state=random_state,
+            min_cls_count_train=1,
+        )
+        return X, y
+
+    # TODO: Should we fillna 0? what about -1 or mean? Does TabPFN automatically fill missing values?
```

Comment:
TabPFNClassifier can handle nans and fills them more or less with the mean of the feature. So, I think it is best to just hand the nans over.

Reply:
Great! I did a quick test and it appears to work without doing the `fillna`, with comparable performance, so I've updated the PR to not fillna and instead pass the NaNs to TabPFN directly.